File: com.observerpattern/python/WeatherApp.py
from WeatherStation import WeatherStation
from PhoneDisplay import PhoneDisplay
from TVDisplay import TVDisplay
from Observer import Observer
from Subject import Subject

def main() -> None:
    weather_station: Subject = WeatherStation()
    phone_display: Observer = PhoneDisplay()
    tv_display: Observer = TVDisplay()

    weather_station.add_observer(phone_display)
    weather_station.add_observer(tv_display)
    
    print(weather_station.observers)
    
    weather_station.set_weather("Sunny")
    
if __name__ == "__main__":
    main()

File: Observer/java/WeatherApp.java
package Observer.java;



// Usage Class
public class WeatherApp {
    public static void main(String[] args) {
        WeatherStation weatherStation = new WeatherStation();

        Observer phoneDisplay = new PhoneDisplay();

        weatherStation.addObserver(phoneDisplay);

        // Simulating weather change
        weatherStation.setWeather("Sunny");

        // Output:
        // Phone Display: Weather updated - Sunny
    }
}